src/LearningObjects/LearningObjectInteractor.ts

```
@@ -91,29 +92,31 @@ export async function updateLearningObject(params: {
       username: params.user.username,
     });
   }
-
+  let error;
   try {
-    await hasLearningObjectWriteAccess(
-      params.user,
-      // TODO: fetch collection
-      'collection',
-      params.dataStore,
-      params.id,
-    );
-    const updates: LearningObjectUpdates = sanitizeUpdates(params.updates);
-    validateUpdates({
-      id: params.id,
-      updates,
-    });
-    updates.date = Date.now().toString();
-    await params.dataStore.editLearningObject({
-      id: params.id,
-      updates,
-    });
-  } catch (e) {
-    return Promise.reject(`Problem updating Learning Object. ${e}`);
+      const hasAccess = await hasLearningObjectWriteAccess(params.user, params.dataStore, params.id); 
+      if(hasAccess) {
+          const updates: LearningObjectUpdates = sanitizeUpdates(params.updates);
+          validateUpdates({
+            id: params.id,
+            updates,
+          });
+          updates.date = Date.now().toString();
+          await params.dataStore.editLearningObject({
+            id: params.id,
+            updates,
+          });
+      } else {
+        error = new Error('User does not have authorization to perform this action');
+        reportError(error);
+        return Promise.reject(error);
+      }
+    } catch (e) {
+      error = new Error(`Problem updating learning object ${params.id}. ${e}`);
```

However, in this case we do want to report an error, as we have hit a catch block. The specific error that we want to report is in fact `e`, the error that was thrown to get us to this point. We want to absorb that error into our reporting service, and generate a new error message that we reject back to the consumer of the service. The block should look something like this:

```
} catch(e) {
  reportError(e);
  return Promise.reject(new Error(`Problem updating learning object ${params.id}.'));
}
```

You'll notice that in this scenario, the error that was thrown is not added to what is rejected. We contain it to this layer of the application, report it, and then generate something scoped for our consumer.